hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+            If 'energy', calibrate the X-ray line energy.
+            If 'sub_weight', calibrate the ratio between the main line
+            alpha and the other sub-lines of the family
+            If 'width', calibrate the X-ray line width.
+        xray_lines: list of str or 'all'
+            The Xray lines. If 'all', fit all lines
+        bounds: float
+            for 'energy' and 'width' the bound in energy, in eV
+            for 'sub_weight' Bound the height of the peak to fraction of
+            its height
+        kind : {'single', 'multi'}
+            If 'single' fit only the current location. If 'multi'
+            use multifit.
+        **kwargs : extra key word arguments
+            All extra key word arguments are passed to fit or
+            multifit, depending on the value of kind.
+        """
+
+        if calibrate == 'energy':
+            bound = (bound / eV2keV) * self.units_factor
+            free = self.free_xray_lines_energy
+            fix = self.fix_xray_lines_energy
+        elif calibrate == 'sub_weight':
+            free = self.free_sub_xray_lines_weight
+            fix = self.fix_sub_xray_lines_weight
+        elif calibrate == 'width':
+            bound = (bound / eV2keV) * self.units_factor
+            free = self.free_xray_lines_width
+            fix = self.fix_xray_lines_width
+
+        free(xray_lines=xray_lines, bound=bound)
+        if kind == 'single':
+            self.fit(bounded=True, fitter='mpfit', **kwargs)
+        elif kind == 'multi':
+            self.multifit(bounded=True, fitter='mpfit', **kwargs)
+        fix(xray_lines=xray_lines)
+
+    def get_lines_intensity(self,
+                            xray_lines=None,
+                            plot_result=False,
+                            **kwargs):
+        """
+        Return the fitted intensity of the X-ray lines.
+
+        Return the area under the gaussian corresping to the X-ray lines
+
+        Parameters
+        ----------
+        xray_lines: list of str or None or 'from_metadata'
+            If None, all main X-ray lines (alpha)
+            If 'from_metadata', take the Xray_lines stored in the `metadata`
+            of the spectrum. Alternatively, provide an iterable containing
+            a list of valid X-ray lines symbols.
+        plot_result : bool
+            If True, plot the calculated line intensities. If the current
+            object is a single spectrum it prints the result instead.
+        kwargs
+            The extra keyword arguments for plotting. See
+            `utils.plot.plot_signals`
+
+        Returns
+        -------
+        intensities : list
+            A list containing the intensities as Signal subclasses.
+
+        Examples
+        --------
+        >>> m.multifit()
+        >>> m.get_lines_intensity(["C_Ka", "Ta_Ma"])
+        """
+        from hyperspy import utils
+        intensities = []
+        if xray_lines is None:
+            xray_lines = []
```

Comment:
`xray_lines = [component.name for component in self.xray_lines]`

Will be quicker and more pythonic